spring-boot-project/spring-boot-tools/spring-boot-buildpack-platform/src/main/java/org/springframework/boot/buildpack/platform/docker/transport/HttpClientTransport.java

```diff
@@ -146,14 +146,35 @@ private Response execute(HttpUriRequest request) {
 		try {
 			ClassicHttpResponse response = this.client.executeOpen(this.host, request, null);
 			int statusCode = response.getCode();
+
 			if (statusCode >= 400 && statusCode <= 500) {
 				byte[] content = readContent(response);
-				response.close();
+
+				if (statusCode == 407) {
+					response.close();
```

Comment:
This looks wrong to me as the response will now only be closed when the status code is 407.